Tasks/Maven/CodeAnalysis/mavencodeanalysis.ts

```diff
@@ -67,28 +65,42 @@ export function uploadCodeAnalysisResults():void {
 }
 
 // Returns the names of any enabled code analysis tools, or empty array if none.
-function getEnabledCodeAnalysisTools():string[] {
-    var result:string[] = [];
+function getEnabledCodeAnalysisTools(): Set<string> {
```

Comment:
ECMA6 has a language-level Set implementation, which is far more appropriate in this case.